meson.build

```diff
@@ -30,18 +30,91 @@ add_project_arguments(cxx.get_supported_arguments(extra_flags), language: 'cpp')
 
 conf = configuration_data()
 conf.set_quoted('PACKAGE_VERSION', meson.project_version())
-conf.set_quoted('DATADIR', join_paths(get_option('prefix'), get_option('datadir')))
+conf.set_quoted('NP_DATADIR', join_paths(get_option('prefix'), get_option('datadir')))
 config_h = configure_file(output : 'config.h', configuration : conf)
 
 meson.add_devenv({'NETPANZER_DATADIR': meson.project_source_root()})
 
-deps = [
-  dependency('SDL2'),
-# SDL2 is case sensitive for the ttf library
-  dependency('SDL2_ttf'),
-  dependency('SDL2_mixer'),
-  dependency('physfs')
-]
+deps = []
+
+if host_machine.system() == 'windows'
+    deps += cxx.find_library('ws2_32')
+endif
+
+# this won't use fallback defined in subprojects/lua.wrap
+lua_dep = dependency('lua-5.1', required: false)
+if not lua_dep.found()
+  lua_dep = cc.find_library('lua', has_headers: 'lua.h', required: false)
+  if not lua_dep.found()
+    # This will use the fallback
+    lua_dep = dependency(
+      'lua-5.1',
+      required: false,
+      fallback: 'lua',
+      default_options: ['default_library=static']
+      )
+  endif
+endif
+deps += lua_dep
+
+phyfs_dep = dependency('physfs', required: false)
+if not phyfs_dep.found()
+  phyfs_dep = cc.find_library('physfs', has_headers: 'physfs.h', required: false)
+  if not phyfs_dep.found()
+    # This will use the fallback
+    phyfs_dep = dependency(
+      'physfs',
+      required: false,
+      fallback: 'physfs',
+      default_options: ['default_library=static']
+      )
+  endif
+endif
+deps += phyfs_dep
+
+sdl2_mixer_dep = dependency('SDL2_mixer', required: false)
+if not sdl2_mixer_dep.found()
+  sdl2_mixer_dep = cc.find_library('SDL2_mixer', has_headers: 'SDL_mixer.h', required: false)
+  if not sdl2_mixer_dep.found()
+    # This will use the fallback
+    sdl2_mixer_dep = dependency(
+      'sdl2_mixer',
+      required: false,
+      fallback: 'sdl2_mixer',
+      default_options: ['default_library=static']
+      )
+  endif
+endif
+deps += sdl2_mixer_dep
+
+sdl2_ttf_dep = dependency('SDL2_ttf', required: false)
+if not sdl2_ttf_dep.found()
+  sdl2_ttf_dep = cc.find_library('SDL2_ttf', has_headers: 'SDL_ttf.h', required: false)
+  if not sdl2_ttf_dep.found()
+    # This will use the fallback
+    sdl2_ttf_dep = dependency(
+      'sdl2_ttf',
+      required: false,
+      fallback: 'sdl2_ttf',
+      default_options: ['default_library=static']
+      )
+  endif
+endif
+deps += sdl2_ttf_dep
+
+sdl2_dep = dependency('SDL2', required: false)
+if not sdl2_dep.found()
+  sdl2_dep = cc.find_library('SDL2', has_headers: 'SDL.h', required: false)
+  if not sdl2_dep.found()
+    # This will use the fallback
+    sdl2_dep = dependency(
+      'sdl2',
+      required: false,
+      fallback: 'sdl2',
+      default_options: ['default_library=static'])
+  endif
+endif
+deps += sdl2_dep
```

Comment:
I think that you also want to look up sdl2 before the sdl2_* dependencies. Otherwise, sdl2 itself might be internally looked up by SDL2_ttf or SDL2_mixer, and meson will already have a cached dependency which doesn't use your default_options.